Products/PleiadesEntity/browser/attestations.py

```diff
@@ -176,35 +177,60 @@ def rows(self, locations):
-            else:
-                status = u''
-            innerHTML = [
-                u'<li id="%s_%s" class="placeChildItem Location" title="%s">' % (
-                    ob.getId(),
-                    where_tag,
-                    self.snippet(ob) + "; " + ob.Description().decode("utf-8"),
-                ),
-                self.prefix(ob),
-                link,
-                self.postfix(ob),
-                status,
-                u'</li>',
-            ]
-            output.append(u"\n".join(innerHTML))
+        location_types = [
+            "associated_modern",
+            "relocated_modern",
+            "central_point",
+            "legacy",
+            "representative",
+        ]
+        labeled_locations = defaultdict(set)
+        for _, ob, _ in sorted(locations, reverse=False):
+            current_location_types = ob.getLocationType()
+            found = False
+            for location_type in location_types:
+                if location_type in current_location_types:
+                    labeled_locations[location_type].add(ob)
+                    found = True
+                    break
+            if not found:
+                labeled_locations['representative'].add(ob)
+        for location_type in location_types:
+            if not labeled_locations[location_type]:
+                continue
+            output.append("<li><b>%s</b>\n<ul>" % location_type.replace('_', ' ').title())
+            for ob in labeled_locations[location_type]:
+                review_state = wftool.getInfoFor(ob, 'review_state')
+                item = ob.Title().decode('utf-8')
+                if 'copy' in ob.getId():
+                    item += u" (copy)"
+                if checkPermission('View', ob):
+                    link = u'<a class="state-%s" href="%s">%s</a>' % (
+                        review_state, ob.absolute_url(), item)
+                else:
+                    link = u'<span class="state-%s">%s</span>' % (
+                        review_state, item)
+                if review_state != 'published':
+                    user = credit_utils.user_in_byline(ob.Creator())
+                    status = u' [%s by %s]' % (review_state, user['fullname'].decode('utf-8'))
+                else:
+                    status = u''
+                innerHTML = [
+                    u'<li id="%s_%s" class="placeChildItem Location" title="%s">' % (
+                        ob.getId(),
+                        where_tag,
+                        self.snippet(ob) + "; " + ob.Description().decode("utf-8"),
+                    ),
+                    self.prefix(ob),
+                    link,
+                    self.postfix(ob),
+                    status,
+                    u'</li>',
+                ]
+                output.append(u"\n".join(innerHTML))
+            output.append(u"</li>")
+        if any(labeled_locations.items()):
+            output.append(u"<ul>")
```

Comment:
Should this be a `</ul>`? I'm seeing some weird HTML in the [source on staging](http://pleiades.jazkarta.com/places/868552020), like:

```
    <label>Locations:</label>
    <ul class="placeChildren">
      <li>
        <b>Representative</b>
       <ul>
         <li id="osm-way-563864846_where" class="placeChildItem Location" title="LineString; unspecified date range; Location based on OpenStreetMap">
            <a class="state-published" href="http://pleiades.jazkarta.com/places/868552020/osm-way-563864846">OSM Way 563864846</a>

            (unspecified date range)

         </li>
         <li id="central-point_where" class="placeChildItem Location" title="Point; unspecified date range; Location based on OpenStreetMap">
           <a class="state-published" href="http://pleiades.jazkarta.com/places/868552020/central-point">Central Point</a>

           (unspecified date range)

         </li>
       </ul>
     </li>
  <ul></ul>
```

That last line should probably just be a single `</ul>` I think.

Reply:
Fixed